packages/data-explorer/src/components/dataView/TableRow.vue

```diff
@@ -1,28 +1,93 @@
 <template>
-  <tr>
-    <td v-if="isShop"><shopping-button :isSelected="isSelected" :id="id"></shopping-button></td>
-    <td v-else>
-      <div v-if="isEditable" class="btn-group" role="group" aria-label="row actions">
-        <button
-          class="btn btn-sm btn-link"
-          role="button"
-          @click="$eventBus.$emit('delete-item', id)"
-          >
-          <font-awesome-icon icon="trash"></font-awesome-icon>
-        </button>
-        <a
-          class="btn btn-sm btn-link"
-          role="button"
-          :href="'/plugin/data-row-edit/' + tableName + '/' + id">
-          <font-awesome-icon icon="edit"></font-awesome-icon>
-        </a>
-      </div>
-    </td>
-    <td v-for="(column, index) in visibleColumns" :key="index">
+  <tr :class="{'row-selected': isSelected}">
+    <th v-if="showSelected" scope="col"><shopping-button :isSelected="isSelected" :id="id"></shopping-button></th>
+    <th v-else class="text-nowrap" :id="'table-row-'+id" scope="col">
+      <input v-if="isEditable" class="form-check-input" type="checkbox" :checked="isSelected" :value="id" @click="$emit('toggleSelectedItemsHandler', id)">
+      <a
+        v-if="isEditable"
+        v-b-tooltip.hover.bottom="'edit'"
+        class="btn btn-sm text-secondary pl-2"
+        role="button"
+        :href="'/plugin/data-row-edit/' + tableName + '/' + id">
+        <font-awesome-icon icon="edit"></font-awesome-icon>
+      </a>
+      <a
+        v-b-tooltip.hover.bottom="'details'"
+        class="btn btn-sm text-secondary"
+        role="button"
+        :href="`/menu/main/dataexplorer/details/${tableName}/${id}`">
+        <font-awesome-icon icon="search"></font-awesome-icon>
+      </a>
+      <button
+        v-if="isEditable"
+        v-b-tooltip.hover.bottom="'delete'"
+        class="btn btn-sm text-secondary"
+        role="button"
+        @click="$eventBus.$emit('delete-item', id)"
+      >
+        <font-awesome-icon icon="trash"></font-awesome-icon>
+      </button>
+    </th>
+    <td v-for="(column, index) in visibleColumns" :key="index" class="text-nowrap text-truncate mg-data-column">
       {{rowData[column.name]}}
     </td>
   </tr>
 </template>
+<style scoped>
+  .row-selected{
+    background-color: var( --tertiary );
+  }
+  /**
+   * Make actions column stick
+   */
+  th {
+    text-align: center;
+    position: -webkit-sticky; /* for Safari */
+    position: sticky;
+    left: 0;
+    background-color: #f7f7f7;
+    border-right: 1px #dee2e6 solid;
+  }
+  /**
+   * Fix borders in the first column
+   */
+  th:after,
+  th:before {
+    content: '';
+    position: absolute;
+    top: 0;
+    height: 100%;
+  }
+  th:before {
+    left: -1px;
+    border-left: 1px solid #dee2e6;
+  }
+  th:after {
+    right: -1px;
+    border-right: 1px solid #dee2e6;
+  }
+
+  /**
+  * Style row actions
+  */
+  .mg-data-column {
+    max-width: 8rem;
```

Comment:
replace with table layout ( for now)

Reply:
this will need to stay, we don't want extreem long column's (like the type test text field)